develop-docs/relay/transaction-span-ratelimits.mdx

```diff
@@ -0,0 +1,230 @@
+| **Transaction Indexed** | ❌                  | ✅                 | ✅           | ✅          |
+| **Span**                | ❌                  | ❌                 | ❌           | ❌          |
+| **Span Indexed**        | ✅                  | ✅                 | ❌           | ✅          |
+
+
+## Outcomes
+
+Outcomes must be generated in Relay for every span and transaction which is dropped. Usually a dropped
+span/transaction results in outcomes for their respective total and indexed category, refer to
+[The Indexed Outcome Category](/application/dynamic-sampling/outcomes/) for details.
+
+This is straight forward for all indexed rate limits, they only drop the payload which results in a single
+negative outcome in the indexed category of the item. A `transaction_indexed` rate limit does not
+cause any spans to be dropped and vice versa.
+
+A span quota and the resulting span rate limit is also trivial for standalone spans received by Relay,
+the standalone span is dropped, and a single outcome is generated.
+
+### Transaction Outcomes
+
+Transactions are containers for spans until they are extracted in Relay. This span extraction can happen at any
+Relay stage: customer managed, PoP-Relay or Processing-Relay. Until spans are extracted from Relay, a dropped transaction
+should count the contained spans and generate an outcome with the contained span quantity + 1, for the segment span
+which would be generated from the transaction itself.
+
+<Note>
+While it is desirable to have span counts correctly extracted from dropped transactions, it may not be feasible
+to do so at any stage of the processing pipeline. For example, it may not be possible to do so (malformed transactions)
+or simply too expensive to compute.
+</Note>
+
+After spans have been extracted, the transaction is no longer a container of span items and just represents itself,
+thus, a dropped transaction with spans already extracted only generates outcomes for the total transactions and
+indexed transaction categories.
+
+<Note>
+Span quotas are equivalent to transactions quotas, so all the above also applies for a span quota.
+</Note>
+
+
+## Examples
+
+### Example: Span Indexed Quota
+
+**Quota**:
+
+```json
+{
+  "categories": ["span_indexed"],
+  "limit": 0
+  // ...
+}
+```
+
+**Transaction**:
+
+```json
+{
+  "type": "transaction",
+  "spans": [
+      { .. },
+      { .. },
+      { .. }
+  ],
+  // ...
+}
+```
+
+An envelope containing a transaction with 3 child spans generates 4 outcomes for rate limited spans in the
+`spans_indexed` category. 1 count for the generated segment span from the transaction and 3 counts for the
+contained spans. The transaction itself will still be ingested.
+
+**Ingestion**:
+
+| Transaction Payload | Transaction Metrics | Span Payload | Span Metrics |
+| ------------------- | ------------------- | ------------ | ------------ |
+| ✅                  | ✅                  | ❌           | ✅           |
```

Comment:
Question: the reason we have span metrics here is that we emit the metric of the transaction that was ingested and treat it as a span metric besides a transaction metric?

Reply:
It goes back to the table defined above, for an indexed quota we **only** drop the payload but keep normal processing going (also produce metrics).

This in isolation (or current AM3 without sampling) may seem a bit weird, but in combination with dynamic sampling it makes more sense. Consider the case dynamic sampling has sampled a payload (we ingest metrics + payload), but then afterwards the abuse quota for indexed payloads forces us to filter the payload, we still want to ingest the metric and not drop it. Then there is no difference whether something is sampled or not, otherwise everything that is not sampled is ingested as metrics, but the few that are sampled are completely dropped.

I also try to explain this a bit in `Should I ever use an indexed category in a Quota?` when you should actually use and indexed quota and the answer is pretty much never, so this example here is a very rare case (or should be).

Reply:
Ah nvm, I misread it was not `span` but `indexed_span`. Thanks!